lib/core/src/Cardano/Wallet/Primitive/Types/UTxOIndex/Internal.hs

```diff
@@ -343,6 +346,12 @@ selectRandom u selectionFilter =
 -- Utilities
 --------------------------------------------------------------------------------
 
+entriesWithAdaOnly :: UTxOIndex -> Set TxIn
+entriesWithAdaOnly u = Map.foldl'
+    (Set.difference)
+    (Map.keysSet $ utxo u)
+    (NonEmptySet.toSet <$> index u)
```

Comment:
This works because the index does not contain any Ada since our `AssetId` data-type does not capture Ada but only other foreign assets.

Reply:
> This works because the index does not contain any Ada since our `AssetId` data-type does not capture Ada but only other foreign assets.

Precisely.

Note that the index refers to entries that **include** a given non-ada asset, but those entries might also include other non-ada assets too.

If we wanted to go further and optimize this, we could also cache this "ada-only" set as an extra field in `Index`. But it's probably not worth it at the moment.